crates/sp-domains/src/fraud_proof.rs

```diff
@@ -468,3 +468,10 @@ pub fn operator_block_rewards_final_key() -> Vec<u8> {
     frame_support::storage::storage_prefix("OperatorRewards".as_ref(), "BlockRewards".as_ref())
         .to_vec()
 }
+
+sp_api::decl_runtime_apis! {
+    /// API related to execution receipts stored in runtime
+    pub trait ExecutionReceiptApi<DomainNumber: Encode + Decode, DomainHash: Encode + Decode> {
+        fn get_execution_receipt_by_hash(hash: ReceiptHash) -> Option<ExecutionReceipt<NumberFor<Block>, Block::Hash, DomainNumber, DomainHash, Balance>>;
+    }
+}
```

Comment:
I think this runtime api can fit into `DomainsApi` and perhaps just `get_execution_receipt` or `execution_receipt` cc @vedhavyas